ttnn/cpp/ttnn/operations/ccl/ccl_common.cpp

```diff
@@ -713,22 +712,21 @@ std::vector<tt_xy_pair> RingReduceScatterWrappedTensorSlicer::create_worker_slic
     std::size_t max_slice_size_in_tiles = max_slice_size_in_pages;
 
     // Assign slices by assuming that the input tensor is flattened into a 1D Shape
-    std::size_t optim_worker_slice_len_tiles = ceil(total_num_tiles / num_workers); // Ceil so that the remainder worker will have a smaller slice
+    std::size_t optim_worker_slice_len_tiles = ((total_num_tiles - 1) / num_workers) + 1; // Ceil so that the remainder worker will have a smaller slice
 
     if (max_slice_size_in_tiles < optim_worker_slice_len_tiles) { // Each worker will have a full slice
         for (uint32_t w = 0; w < num_workers; ++w) {
             worker_slice_shapes.emplace_back(max_slice_size_in_tiles, 1);
         }
     } else { // Each worker will only have one slice
-        uint32_t remainder_worker_len_tiles = total_num_tiles % optim_worker_slice_len_tiles;
-
+        size_t base_tiles_per_worker = total_num_tiles / num_workers;
```

Comment:
This should be a constant

Reply:
I used to const all my local variables everywhere possible but found that it added noise to code in some cases. I usually am more selective when I apply it to local variables now.

I don't feel strongly one way or another on this.

From google style guide that we follow
"Using const on local variables is neither encouraged nor discouraged."

If these were constexprable though then I would always mark constexpr

Reply:
Your correct, a good optimizer should notice this at least in the local scope, I guess to many years of Vivado taught me not to rely on the optimizer that much